controllers/partitionjob_controller.go

```diff
@@ -175,21 +167,22 @@ func (r *PartitionJobReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		l.Info("Scaling up pods", "Currently available", numAvailableReplicas, "Required replicas", partitionJob.Spec.Replicas)
 		diff := *partitionJob.Spec.Partitions - partitionJob.Status.UpdatedReplicas
 
+		newUpdatedReplicas := partitionJob.Status.UpdatedReplicas
+
 		var pod *corev1.Pod
-		if updatedRevisionData != nil {
-			if diff > 0 {
-				//create pod with new revision
-				template := utils.RawToTemplate(updatedRevisionData)
-				pod = utils.CreateNewPod(partitionJob, template)
+		if diff > 0 {
+			//create pod with new revision
+			template := utils.RawToTemplate(updatedRevision.Data.Raw)
+			pod = utils.CreateNewPod(partitionJob, template)
+			newUpdatedReplicas++
```

Comment:
should we not do the same with currentReplicas in the else statement below?

Reply:
We can but we are not using current replicas anywhere and its value gets updated start of the new loop. Good catch. Let me do it for consistency's sake